cmd/atlas/internal/cmdapi/schema.go

```diff
@@ -445,19 +447,37 @@ flag.
 }
 
 func schemaInspectRun(cmd *cobra.Command, _ []string, flags schemaInspectFlags) error {
-	client, err := sqlclient.Open(cmd.Context(), flags.url)
+	var (
+		ctx = cmd.Context()
+		dev *sqlclient.Client
+	)
+	if flags.devURL != "" {
+		var err error
+		dev, err = sqlclient.Open(ctx, flags.devURL)
+		if err != nil {
+			return err
+		}
+		defer dev.Close()
+	}
+	r, err := stateReader(ctx, &stateReaderConfig{
+		urls:    []string{flags.url},
+		dev:     dev,
+		vars:    GlobalFlags.Vars,
+		schemas: flags.schemas,
+		exclude: flags.exclude,
+	})
 	if err != nil {
 		return err
 	}
-	defer client.Close()
-	schemas := flags.schemas
-	if client.URL.Schema != "" {
-		schemas = append(schemas, client.URL.Schema)
+	defer r.Close()
+	client, ok := r.Closer.(*sqlclient.Client)
```

Comment:
Keep it close to its usage below? (After reading the state).

Reply:
Humm. Usually, we keep the call to close() close to the open function, no?

Reply:
I meant the definition of the `client` variable.